crates/ruma-events/src/sticker.rs

```diff
@@ -24,13 +50,31 @@ pub struct StickerEventContent {
     /// Metadata about the image referred to in `url` including a thumbnail representation.
     pub info: ImageInfo,
 
-    /// The URL to the sticker image.
-    pub url: OwnedMxcUri,
+    /// The media source of the sticker image.
+    #[serde(flatten)]
+    pub source: StickerMediaSource,
 }
 
 impl StickerEventContent {
     /// Creates a new `StickerEventContent` with the given body, image info and URL.
+    #[cfg(not(feature = "compat-encrypted-stickers"))]
     pub fn new(body: String, info: ImageInfo, url: OwnedMxcUri) -> Self {
-        Self { body, info, url }
+        Self { body, info, source: StickerMediaSource::Plain(url.clone()) }
+    }
+    /// Creates a new `StickerEventContent` with the given body, image info and URL.
+    #[cfg(feature = "compat-encrypted-stickers")]
+    pub fn new(body: String, info: ImageInfo, url: OwnedMxcUri) -> Self {
+        Self { body, info, source: StickerMediaSource::Plain(url.clone()) }
+    }
+
+    /// Creates a new `StickerEventContent` with the given body, image info, URL, and media source.
+    #[cfg(not(feature = "compat-encrypted-stickers"))]
+    pub fn from_source(body: String, info: ImageInfo, source: StickerMediaSource) -> Self {
+        Self { body, info, source }
+    }
+    /// Creates a new `StickerEventContent` with the given body, image info and URL.
+    #[cfg(feature = "compat-encrypted-stickers")]
+    pub fn from_source(body: String, info: ImageInfo, url: OwnedMxcUri) -> Self {
+        Self { body, info, source: StickerMediaSource::Plain(url.clone()) }
```

Comment:
The first method should be enough, and it doesn't need the feature flag because it would work in both cases. Also I believe that the Rust convention would be to name it `with_source`.